### ISSUE TITLE
Tailscale on enigna

### ISSUE THREAD

**Issue by estimadarocha** [tailscale/tailscale#1291]
I move recently from zerotier to tailscale but i am fighting the missing integration on https://github.com/oe-alliance/oe-alliance-core so that it could be integrated and installed on Enigma2 stb. I did ask on some forums about those devices and the feature will be very welcome. 

Could someone look into it?

I can help on testing. 

Thanks

**Comment by bradfitz** [comment]
What is it that you need? That link doesn't explain anything.

**Comment by estimadarocha** [comment]
oe-alliance-core is the core of a large group of  teams that make images for enigma2 stb's.

i think this was the initial comit for zerotier... i hope it helps somehow...
https://github.com/oe-alliance/oe-alliance-core/commit/cf88be1484d6972daacceee89579f8f4b72de355

**Comment by bradfitz** [comment]
Gotcha, thanks. Retitled & labeled.

**Comment by estimadarocha** [comment]
Big big thanks... Do you think it's something hard to achieve?

**Comment by bradfitz** [comment]
It doesn't look hard but it might be time consuming. Especially as we don't have that hardware or a simulated environment for testing.

**Comment by estimadarocha** [comment]
I can test all you need and help as much as I can

**Comment by bradfitz** [comment]
That's where the time consuming part comes in. High latency turn around asking somebody else to do things a hundred times in a row whereas a hundred explorative missteps locally with near zero latency is painless. 

It'd be best for somebody with the appropriate dev environment to do this.

**Comment by estimadarocha** [comment]
I can get you remote access to one of these stbs... Will it help?

**Comment by bradfitz** [comment]
Yes, that would. I still can't promise any timeline though as we're all constantly busy on other things and need to prioritize. (Personally I want to see us polish up our Synology packages, as there are quite a number of users)

**Comment by estimadarocha** [comment]
I did use it also on competing company (qnap) working very well indeed. Regarding the numbers I am pretty sure there's more users on Enigma2 devices than on qnap and synology together. If I have time I will also try to setud a build environment of alliance core.

**Comment by estimadarocha** [comment]
ok so the support for tailscale on oe-alliance-core is now added.

I tried last night the package but i am having some troubles. I can't ping peer to peer with normal ping! is it normal?

I created also another peer on android cell phone and i can see peer services, but even that is not working very good, is like something missing on routes.

Thanks

**Comment by Balumis** [comment]
> ok so the support for tailscale on oe-alliance-core is now added.
> 
> I tried last night the package but i am having some troubles. I can't ping peer to peer with normal ping! is it normal?
> 
> I created also another peer on android cell phone and i can see peer services, but even that is not working very good, is like something missing on routes.
> 
> Thanks

Hi, do you have stable version, how it's working now?

**Comment by DentonGentry** [comment]
The steps at https://jungle--team-com.translate.goog/tailscale-conecta-tu-receptor-enigma2-a-una-red-privada-vpn/?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=pl&_x_tr_pto=wapp reportedly works.

If the version of the tailscale binaries is old, replacing /usr/bin/tailscale and /usr/bin/tailscaled should work.
Static binaries can be found at https://pkgs.tailscale.com/stable/#static for many CPU architectures.

**Comment by DentonGentry** [comment]
Relevant for this audience: https://tailscale.com/kb/1207/small-tailscale/

**Comment by adrianmihalko** [comment]
Anyone got this working? I can't get it to work. I think the tun device is not created. After running "tailscaled & tailscale up", should I see the tun interface in the ifconfig output? The tun module is loaded (modprobe tun), I see devices in tailscale status.

**Comment by DentonGentry** [comment]
On many Linux platforms, /dev/net/tun needs to be explicitly created: `mknod /dev/net/tun c 10 200`
It doesn't appear dynamically after loading the module.

**Comment by schmierlappe** [comment]
> The steps at https://jungle--team-com.translate.goog/tailscale-conecta-tu-receptor-enigma2-a-una-red-privada-vpn/?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=pl&_x_tr_pto=wapp reportedly works.
> 
> If the version of the tailscale binaries is old, replacing /usr/bin/tailscale and /usr/bin/tailscaled should work. Static binaries can be found at https://pkgs.tailscale.com/stable/#static for many CPU architectures.

but how it should look when you want to connect two devices with running in userspace-networking mode. I think they cant see each other or?

**Comment by DentonGentry** [comment]
userspace-networking means that regular Linux socket apps won't be able to use the tailnet. Apps must use a SOCKS5 or HTTP proxy to be able to connect to the tailnet.
https://tailscale.com/kb/1112/userspace-networking
